NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs

```diff
@@ -21,6 +22,21 @@ public OperationNameTelemetryInitializer(IHttpContextAccessor httpContextAccesso
         /// <inheritdoc />
         protected override void OnInitializeTelemetry(HttpContext platformContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
         {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException(nameof(telemetry));
+            }
+
+            if (platformContext == null)
+            {
+                throw new ArgumentNullException(nameof(platformContext));
```

Comment:
This now throws even if requestTelemetry.name is not empty and execution does not reach platformContext branch.

Reply:
moved into if